apm-agent-plugins/apm-hibernate-search-plugin/apm-hibernate-search-plugin-6_x/src/main/java/co/elastic/apm/agent/hibernate/search/v6_x/HibernateSearch6Instrumentation.java

```diff
@@ -0,0 +1,82 @@
+package co.elastic.apm.agent.hibernate.search.v6_x;
+
+import java.util.Collection;
+import java.util.Collections;
+
+import javax.annotation.Nullable;
+
+import co.elastic.apm.agent.bci.ElasticApmInstrumentation;
+import co.elastic.apm.agent.bci.VisibleForAdvice;
+import co.elastic.apm.agent.hibernate.search.HibernateSearchConstants;
+import co.elastic.apm.agent.impl.transaction.Span;
+import co.elastic.apm.agent.impl.transaction.TraceContextHolder;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.hibernate.search.backend.lucene.search.query.impl.LuceneSearchQueryImpl;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+public class HibernateSearch6Instrumentation extends ElasticApmInstrumentation {
+
+    @Override
+    public Class<?> getAdviceClass() {
+        return Hibernate6ExecuteAdvice.class;
+    }
+
+    @Override
+    public ElementMatcher<? super TypeDescription> getTypeMatcher() {
+        return named("org.hibernate.search.backend.lucene.search.query.impl.LuceneSearchQueryImpl");
+    }
+
+    @Override
+    public ElementMatcher<? super MethodDescription> getMethodMatcher() {
+        return named("fetch");
```

Comment:
What about other query methods in `org.hibernate.search.engine.search.query.SearchFetchable`?

Reply:
That depends how far the support of the 6.x version should go since it still is in alpha and there could be still some changes made.

Reply:
Is there a difference in expected stability between the `fetch` methods and the others? Is there a reason why they are preferable?

Reply:
As I'm not involved in the development I personally would expect at the current state that everything could be changed. The `fetch` method is just the equivalent of the `list` method I traced for Hibernate 5. So no reason to prefer it over the other methods when adding tracing support.

Reply:
I now trace all methods from the `SearchFetchable` interface.